inc/Engine/Optimization/RUCSS/Warmup/Scanner.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace WP_Rocket\Engine\Optimization\RUCSS\Warmup;
+
+use WP_Rocket\Admin\Options;
+use WP_Rocket\Engine\Optimization\ContentTrait;
+
+class Scanner {
+	use ContentTrait;
+
+	/**
+	 * Background Process instance.
+	 *
+	 * @var ScannerProcess
+	 */
+	public $process;
+
+	/**
+	 * Options API instance.
+	 *
+	 * @var Options
+	 */
+	private $options_api;
+
+	/**
+	 * Items for which we get the resources to warmup.
+	 *
+	 * @var array
+	 */
+	public $items = [];
+
+	/**
+	 * Instantiate the class
+	 *
+	 * @param ScannerProcess $process Background process instance.
+	 * @param Options        $options_api Options API instance.
+	 */
+	public function __construct( ScannerProcess $process, Options $options_api ) {
+		$this->process     = $process;
+		$this->options_api = $options_api;
+	}
+
+	/**
+	 * Launches the scanner when activating the RUCSS option
+	 *
+	 * @since 3.9
+	 *
+	 * @param array $old_value Previous values for WP Rocket settings.
+	 * @param array $value     New values for WP Rocket settings.
+	 *
+	 * @return void
+	 */
+	public function start_scanner( $old_value, $value ) {
+		if (
+			! isset( $old_value['remove_unused_css'], $value['remove_unused_css'] )
+			||
+			( $old_value['remove_unused_css'] === $value['remove_unused_css'] )
+			|| 1 !== (int) $value['remove_unused_css']
+		) {
+			return;
+		}
+
+		$this->dispatcher();
+	}
+
+	/**
+	 * Gets the items to scan and dispatch them to the scanner
+	 *
+	 * @since 3.9
+	 *
+	 * @return void
+	 */
+	private function dispatcher() {
+		$this->set_items();
+
+		array_map( [ $this->process, 'push_to_queue' ], $this->items );
+
+		$this->options_api->set( 'scanner_start_time', current_time() );
```

Comment:
I like saving it with parameter `U` to get the timestamp so we can calculate easily how much time elapsed in seconds and comparing them with WP constants like `HOUR_IN_SECONDS`, what do u think ?

Reply:
changed it to use `time()` directly